sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2618,23 +3024,29 @@ export class SearchIndexClient {
     readonly apiVersion: string;
     createAlias(alias: SearchIndexAlias, options?: CreateAliasOptions): Promise<SearchIndexAlias>;
     createIndex(index: SearchIndex, options?: CreateIndexOptions): Promise<SearchIndex>;
+    createKnowledgeAgent(knowledgeAgent: KnowledgeAgent, options?: CreateKnowledgeAgentOptions): Promise<KnowledgeAgent>;
```

Comment:
Can you help me understand this one? `createKnowledgeAgent` takes a `KnowledgeAgent` as a param and returns a `KnowledgeAgent` as a promise'd result. What is different about the input and output?

Reply:
How is it different than `createOrUpdateKnowledgeAgent`? That's where I'd have expected ETag to make sense

Reply:
The response for both includes an ETag and some other service defaults that you wouldn't or might not define in the request body. The `create` method should always fail for existing knowledge agents.